go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/containers/podman/v4
 
-go 1.18
+go 1.21
+
+toolchain go1.21.3
```

Comment:
breaks on f38 and centos stream

Reply:
and without it, it breaks on fedora 39 and rawhide :/

Reply:
looks like all versions are happy now

Reply:
Looking at https://bodhi.fedoraproject.org/updates/?packages=golang , F38 still needs to build against Go 1.20, doesn’t it? If so, I think we should keep running tests (also?) against Go ≤ 1.20 …

… but this is, it seems, required by https://github.com/containers/buildah/pull/5139 (for https://github.com/containers/buildah/pull/5007 which might have been fine with 1.20, not 1.21?).

@lsm5 Do we need to revert Buildah?

Reply:
I am still getting a weird error on F39 with `hack/golangci-lint.sh run` when I vendor buildah.

The root cause is that this command fails:

```
GOROOT=/usr/lib/golang GOPATH=/home/gscrivano/src/gopath GO111MODULE= GOPROXY= PWD=/home/gscrivano/src/gopath/src/github.com/containers/podman go list -tags "apparmor seccomp selinux systemd" -mod=readonly -e -json=Name,ImportPath,Error,Dir,GoFiles,IgnoredGoFiles,IgnoredOtherFiles,CFiles,CgoFiles,CXXFiles,MFiles,HFiles,FFiles,SFiles,SwigFiles,SwigCXXFiles,SysoFiles,TestGoFiles,XTestGoFiles,CompiledGoFiles,Export,DepOnly,Imports,ImportMap,TestImports,XTestImports -compiled=true -test=true -export=true -deps=true -find=false -pgo=off -tags "apparmor seccomp selinux systemd" -mod=readonly -- ./...
```

Of course once I run "go mod tidy" the go.mod is converted to go 1.21